pkg/audit/manager.go

```diff
@@ -401,15 +419,130 @@ func (am *Manager) auditResources(
 					break
 				}
 			}
+			// loop thru all subDirs to review all files for this kind
+			err = am.reviewObjects(ctx, kind, folderCount, nsCache, updateLists, totalViolationsPerConstraint, totalViolationsPerEnforcementAction, timestamp)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
 		}
 	}
+	if len(errs) > 0 {
+		return errs
+	}
+	return nil
+}
 
+func (am *Manager) reviewObjects(ctx context.Context, kind string, folderCount int, nsCache *nsCache,
+	updateLists map[util.KindVersionResource][]auditResult,
+	totalViolationsPerConstraint map[util.KindVersionResource]int64,
+	totalViolationsPerEnforcementAction map[util.EnforcementAction]int64,
+	timestamp string) error {
+	var errs opa.Errors
+	for i := 0; i < folderCount; i++ {
+		subDir := fmt.Sprintf("%s_%d", kind, i)
+		pDir := path.Join(*apiCacheDir, subDir)
+
+		files, err := am.getFilesFromDir(pDir, int(*auditChunkSize))
```

Comment:
While this pages files from the OS, it still aggregates the list in memory before processing. This should be fine unless we expect the list to be very large. Otherwise we could pass in a processing function to be called per page.

Reply:
Or maybe walk the directory and call audit on every file we walk into?

`walk()` is probably preferable, but I don't have an issue with this as-is, since the list of file names is probably much smaller than the list of actual objects.

Reply:
I think the list of file names is pretty small compare to everything else (actual resources) we are caching. We can further optimize this if we see a dramatic improvement later.